src/System.IO.FileSystem/src/System/IO/FileSystem.Windows.cs

```diff
@@ -111,6 +109,24 @@ public static void CreateDirectory(string fullPath)
             // descriptor and set it in he call to CreateDirectory.
             Interop.Kernel32.SECURITY_ATTRIBUTES secAttrs = default;
 
+            if (dirSecurity != null)
+            {
+                secAttrs.nLength = (uint)Marshal.SizeOf(secAttrs);
```

Comment:
Don't ever use `Marshal.SizeOf`. You should only be using it if our struct is not blittable, which would be a problem.